packages/workflows/src/helper/pipe.ts

```diff
@@ -22,25 +22,23 @@ interface PipelineInput {
   compensate?: WorkflowStepMiddlewareInput | WorkflowStepMiddlewareInput[]
 }
 
-export type WorkflowArguments = {
+export type WorkflowArguments<T = any> = {
   container: MedusaContainer
   payload: unknown
-  data: any
+  data: T
   metadata: TransactionMetadata
   context: Context | SharedContext
 }
 
-export type PipelineHandlerResult<T = undefined> = T extends undefined
```

Comment:
you might have to remove this from existing stuff

Reply:
done already 👍 thanks